<a href="https://colab.research.google.com/github/purnawanpp/plants/blob/main/plants.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Cara Menjalankan Software Protein-Ligand ANTSystem (PLANTS) di google colab by Purnawan Pontana Putra**

**Instalasi Software yang dibutuhkan**

In [1]:
!wget https://github.com/purnawanpp/plants/raw/main/plants
!wget https://github.com/purnawanpp/plants/raw/main/spores
!apt install openbabel
!chmod +x /content/plants
!chmod +x /content/spores
!wget https://github.com/purnawanpp/plants/raw/main/dock_gcolab.txt

--2023-03-08 10:18:30--  https://github.com/purnawanpp/plants/raw/main/plants
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/purnawanpp/plants/main/plants [following]
--2023-03-08 10:18:31--  https://raw.githubusercontent.com/purnawanpp/plants/main/plants
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9536847 (9.1M) [application/octet-stream]
Saving to: ‘plants’

plants              100%[===================>]   9.09M  --.-KB/s    in 0.09s   

2023-03-08 10:18:31 (99.7 MB/s) - ‘plants’ saved [9536847/9536847]

--2023-03-08 10:18:31--  https://github.com/purnawanpp/plants/raw/main/spores
Resolvi

**Download Protein PDB ID 3HTB**

In [2]:
!wget https://files.rcsb.org/download/3HTB.pdb

--2023-03-08 10:19:13--  https://files.rcsb.org/download/3HTB.pdb
Resolving files.rcsb.org (files.rcsb.org)... 128.6.158.70
Connecting to files.rcsb.org (files.rcsb.org)|128.6.158.70|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘3HTB.pdb’

3HTB.pdb                [ <=>                ] 161.13K  --.-KB/s    in 0.06s   

2023-03-08 10:19:13 (2.56 MB/s) - ‘3HTB.pdb’ saved [164997]



**Separete Protein and ligand**

In [3]:
!./spores --mode splitpdb 3HTB.pdb


				 SPORES        

	      Structure PrOtonation and REcognition System

			      version 1.3


author: T. ten Brink

scientific contributors: P. Duchstein, J. Gimmler, O. Korb, M. Zerara,
			  T.E. Exner

contact: tim.tenbrink@uni-konstanz.de


molecule number: 1
64 alternative atomic positions have been removed !
splitting 3HTB
225
protein: 0
ligs: 4
water: 660
metal: 0


**Add Hydrogen, charge (gastaiger) dan protonasi pada pH 7.4**

Keterangan:

⚾!obabel: perintah untuk menjalankan program Obabel pada Google Colab Notebook.

⚾-imol2: opsi untuk menentukan format input file, yaitu mol2.

⚾/content/protein.mol2: jalur file input yang akan dikonversi dan dimanipulasi.

⚾-omol2: opsi untuk menentukan format output file, yaitu mol2.

⚾-h: opsi untuk menambahkan hidrogen pada molekul.

⚾-p 7.4: opsi untuk menambahkan muatan pada molekul dengan pH 7.4.

⚾-b: opsi untuk menambahkan informasi ikatan hidrogen.

⚾--ff "gasteiger": opsi untuk menentukan gaya perhitungan muatan, dalam hal ini menggunakan metode Gasteiger.

⚾-O rec.mol2: opsi untuk menentukan nama dan lokasi file output, yaitu rec.mol2




In [24]:
!obabel -imol2 protein.mol2 -omol2 -h -b --ff "gasteiger" -O rec.mol2
!obabel -imol2 ligand_JZ4167_0.mol2 -omol2 -h -b --ff "gasteiger" -O lig.mol2

*** Open Babel Warning  in ReadMolecule
  Failed to kekulize aromatic bonds in MOL2 file (title is 3HTB_protein)

1 molecule converted
1 molecule converted


**Membuat Binding Sitenya atau Gridboxnya**

In [5]:
!./plants --mode bind lig.mol2 rec.mol2



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile



Calculated bindingsite definition (from 1 ligands): 

bindingsite_center 22.7142 -25.2627 -3.283
bindingsite_radius 3.75624



Pindahkan data:

*   bindingsite_center 22.7142 -25.2627 -3.283
*   bindingsite_radius 3.75624

kedalam file **dock_gcolab.txt**

**Simulasi Molecular Docking**

In [25]:
!./plants --mode screen /content/dock_gcolab.txt



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile

PLANTS info: CHO hydrogen found 513 HD2 HIS3131
PLANTS info: CHO hydrogen found 514 HE1 HIS3131
virtual screening progress: 1 of 1
current ligand: /content/lig.mol2 (entry 1)
LIGAND DOFs: 9
PROTEIN DOFs: 0
Simplex dimension: 9
Simplex dimension: 9
iterations : 187
starting optimization ...
problem dimension: 9
ATOMS / s: 8.99545e+06
EVAL / s: 899545
optimization finished after 0.81s
best score: -67.44


total virtual screening time: 0.98s
Ligands skipped due to errors: 0


Setelah proses selesai, jangan lupa klik simbol kotak/file refresh google colab. Hasil simulasi docking dapat dilihat pada folder *results* dengan nama file **bestranking.csv**, lihat energy TOTAL_SCORE merupakan nilai delta G

**Melihat score nilai delta G atau hasil docking**

In [26]:
import pandas as pd
df = pd.read_csv('/content/results/bestranking.csv')
df.head()

LIGAND_ENTRY  TOTAL_SCORE  SCORE_RB_PEN  \
0  3HTB_ligand_2_entry_00001_conf_01     -67.4369      -61.4369   

   SCORE_NORM_HEVATOMS  SCORE_NORM_CRT_HEVATOMS  SCORE_NORM_WEIGHT  \
0             -6.74369                 -31.3014          -0.495182   

   SCORE_NORM_CRT_WEIGHT  SCORE_RB_PEN_NORM_CRT_HEVATOMS  SCORE_NORM_CONTACT  \
0               -13.1075                        -28.5165            -6.74369   

     EVAL      TIME  
0  728357  0.809695

Keterangan:
* LIGAND_ENTRY: Nama atau kode identifikasi ligand yang di-dockingkan.
* TOTAL_SCORE: Skor keseluruhan yang dihasilkan dari docking ligand pada protein target.
* SCORE_RB_PEN: Skor untuk penalti rotamer.
* SCORE_NORM_HEVATOMS: Skor untuk heteroatom berat (dalam hal ini, nilai yang dinormalisasi).
* SCORE_NORM_CRT_HEVATOMS: Skor untuk heteroatom berat yang dinormalisasi dengan koreksi.
* SCORE_NORM_WEIGHT: Skor yang dinormalisasi dengan berat molekul ligand.
* SCORE_NORM_CRT_WEIGHT: Skor yang dinormalisasi dengan koreksi dan berat molekul ligand.
* SCORE_RB_PEN_NORM_CRT_HEVATOMS: Skor untuk rotamer dan heteroatom berat yang dinormalisasi dengan koreksi.
* SCORE_NORM_CONTACT: Skor untuk kontak yang dinormalisasi.
* EVAL: Nilai evaluasi yang mengukur kualitas docking antara ligand dan protein target.
* TIME: Waktu yang dibutuhkan untuk melakukan docking.

**Perhitungan RMSD hasil simulasi (re-docking)**

In [21]:
!obrms -f lig.mol2 /content/results/3HTB_ligand_2_entry_00001_conf_01.mol2 | sed 's/3HTB_ligand_2:3HTB_ligand_2_entry_00001_conf_01/RE-DOCKING/g'

RMSD RE-DOCKING 0.434581
